stablehlo/transforms/StablehloAggressiveSimplification.cpp

```diff
@@ -70,28 +66,19 @@ template <typename Fn>
 static TypedAttr foldBinaryOpIntOrFloat(TypedAttr lhs, TypedAttr rhs,
                                         Fn &&folder) {
   Attribute operands[2] = {lhs, rhs};
-  Type elemTy = getElementTypeOrSelf(cast<TypedAttr>(lhs).getType());
+  Type elemTy = getElementTypeOrSelf(lhs);
 
+  Attribute res;
   if (isa<IntegerType>(elemTy)) {
-    if (Attribute res =
-            constFoldBinaryOp<IntegerAttr, IntegerAttr::ValueType, void>(
-                operands, [&folder](const APInt &lhs, const APInt &rhs) {
-                  return folder(lhs, rhs);
-                })) {
-      return cast<TypedAttr>(res);
-    }
-    return nullptr;
+    res = constFoldBinaryOp<IntegerAttr, IntegerAttr::ValueType, void>(operands,
+                                                                       folder);
   }
-
   if (isa<FloatType>(elemTy)) {
-    if (Attribute res =
-            constFoldBinaryOp<FloatAttr, FloatAttr::ValueType, void>(
-                operands, [&folder](const APFloat &lhs, const APFloat &rhs) {
-                  return folder(lhs, rhs);
-                })) {
-      return cast<TypedAttr>(res);
-    }
-    return nullptr;
+    res = constFoldBinaryOp<FloatAttr, FloatAttr::ValueType, void>(operands,
+                                                                   folder);
+  }
+  if (res) {
+    return cast<TypedAttr>(res);
```

Comment:
Does `res.cast<TypedAttr>();` work? I always forget which support casting as a method

Reply:
revisit other uses of `cast` as well.